File: Ideal.php
<?php
//If the year has or not 1 week more
function days(int $y) {
	return $y==round(round($y*1.242189/7)*7/1.242189)?371:364;
}
class Ideal {
	private $D;//Days since the change of millennium initiating in 0.
	private $y=2000;//Year initiating in 2000
	private $d=6;//Day of the year counting since 0 (Sunday), 6 is Saturday, day that the change of millennium was.
	//The Gregorian 1/1/2000 would be 7/1/2000 in the ideal calendar, a standard I propose.
	//Gregorian, year and day
	//If d>days(y), days pass to some posterior year. If d<1, days pass to some previous year.
	public function __construct($g, $y=2000, $d=7) {
		try { 
			$this->addDays(($g-(int)date("U",mktime(0,0,0,1,1,2000)))/86400);
		}catch(Exception $e) {
			$this->addYears($y-2000);
			$this->addDays($d-7);
		}
	}
	public function addDays(int $n) {
		$this->d+=$n;
		if($n>0) {
			$N=days($this->y);
			while($this->d>=$N) {
				$this->d-=$N;
				$N=days(++$this->y);
			}
		}
		else while($this->d<0) $this->d+=days(--$this->y);
		$this->D+=$n;
	}
	public function addYears(int $n) {
		$a=$this->y+$n;
		if($this->d>363) if(days($a)==364) {
			$this->d-=7;
			$this->D-=7;
		}
		if($n>0) while($this->y<$a) $this->D+=days($this->y++);
		if($n<0) while($this->y>$a) $this->D-=days(--$this->y);
	}
	//Choose among year, month, week, day of the year/month/week.
	//Week and month start at 0 internally and here 1 is added.
	public function __get($f) {
		switch($f) {
			case 'y': return $this->y;
			case 'm': return floor($this->d/28)+1;
			case 'w': return floor($this->d/7)+1;
			case 'd': return $this->d;
			case 'D': return $this->D;
			case 'dm': return $this->d%28+1;
			case 'dw': return $this->d%7+1;
			//Ideal calendar to Gregorian.
			case 'g': return date_add(date_create("2000-01-01"),date_interval_create_from_date_string($this->D.' days'));
		}
		return 0;
	}
}
$i=new Ideal((int)date("U"));
echo date_format($i->g,"Y/m/d").'<br>'.$i->dm." ".$i->m." ".$i->y.'<br>';

File: ideal.vb
'If the year has or not 1 week more
Function days(y As Long) As Integer
    r = Round(Round(y * 1.242189 / 7) * 7 / 1.242189)
    If y = r Then
        days = 371
    Else
        days = 364
    End If
End Function
'Date as number in Excel and version
Function ideal(f As Long, v As Integer) As String
    Dim n, d, a, m As Integer
    Dim y As Long
    'Days since the glorious change of millennium
    '36526 days from 1900 to 2000 in Excel
    a = f - 36526
    'Day of the year counting since 0 (Sunday), 6 is Saturday, day that the change of millennium was.
    d = 6 + a
    y = 2000 'Year initiating in 2000
    If a > 0 Then
        n = days(y)
        While d >= n
            d = d - n
            y = y + 1
            n = days(y)
        Wend
    Else
        While d < 0
            y = y - 1
            d = d + days(y)
        Wend
    End If
    ideal = ""
    'Day of the year, day (1 to 28) and month or day (1 to 7) and week and the year
    If v = 1 Then
        m = 28
    ElseIf v = 2 Then
        m = 7
    End If
    If v = 1 Or v = 2 Then
        ideal = ((d Mod m) + 1) & " " & (Round(d / m) + 1) & " " & y
    ElseIf v = 3 Then
        ideal = d & " " & y
    End If
End Function
'Year and day
Function gregorian(d As Integer, y As Long) As Long
    'The Gregorian 1/1/2000 would be 7/1/2000 in the ideal calendar, a standard I propose.
    gregorian = 36519 + d
    Dim i As Long
    i = 2000
    If y > 2000 Then
        While i < y
            gregorian = gregorian + days(i)
            i = i + 1
        Wend
    End If
    If y < 2000 Then
        While i > y
            i = i - 1
            gregorian = gregorian - days(i)
        Wend
    End If
End Function